src/nupic/regions/PyRegion.py

```diff
@@ -177,6 +177,18 @@ def getOutputElementCount(self, name):
     name: the name of the output
     """
 
+  @not_implemented
+  def getAlgorithmInstance(self):
```

Comment:
I don't think this should be added to PyRegion. Adding it here requires that every region have an "algorithm" which may not always make sense and the return value type will be different in every case. But fine to use the same name in the regions where we choose to implement it as a convention if that makes the API more discoverable.